draft-ietf-dnsop-svcb-https.md

```diff
@@ -750,7 +750,7 @@ pairs are concatenated to form the SvcParamValue.  These pairs MUST exactly
 fill the SvcParamValue; otherwise, the SvcParamValue is malformed.
 
 For "no-default-alpn", the presentation and wire format values MUST be
-empty.
+empty.  When "no-default-alpn" is specified, "alpn" MUST also be specified.
```

Comment:
```suggestion
empty.  When "no-default-alpn" is specified in an RR, 
"alpn" MUST also be specified in-order for the RR 
to be "self-consistent" ({{service-mode}}).
```

Reply:
(to align with #285)

Reply:
:+1:

Reply:
Nit: If I'm remembering the language correctly, being self-consistent is only a SHOULD, so any "in order for the RR to be 'self-consistent'" rules should probably also be a SHOULD, rather than a MUST.  You can create records that violate it, but if you do, you have to expect clients will ignore the RR or RRSet.

Reply:
When not "self-consistent", dropping the RR is a MUST for clients, rejecting the entire RRset is a MAY.

Reply:
I read this as "The record is not self-consistent if it has no-default-alpn but not alpn.".  I think it's fine.  It is perhaps slightly unusual use of normative language ... we could make it a lower-case "must" :).

Reply:
But zone implementors enforcing that records are self-consistent was only a SHOULD, right? Making this a MUST would add an additional strong requirement specific to this rule.

Reply:
To further clarify my point, ""alpn" MUST also be specified in-order for the RR to be "self-consistent" is a rule for the zone implementors to follow.  They're the ones able to make that happen or not.  That rule is then inconsistent with "Zone-file implementations SHOULD enforce self-consistency."

I think either ""alpn" SHOULD also be specified in-order for the RR to be "self-consistent" or ""alpn" must also be specified in-order for the RR to be "self-consistent" would be sufficient to break the inconsistency, but I think the former is slightly clearer.

Reply:
Let's argue in #290.